source/adios2/operator/compress/CompressBZIP2.cpp

```diff
@@ -0,0 +1,231 @@
+/*
+ * Distributed under the OSI-approved Apache License, Version 2.0.  See
+ * accompanying file Copyright.txt for details.
+ *
+ * CompressBZIP2.cpp
+ *
+ *  Created on: Jul 24, 2017
+ *      Author: William F Godoy godoywf@ornl.gov
+ */
+
+#include "CompressBZIP2.h"
+
+#include <cmath>     //std::ceil
+#include <ios>       //std::ios_base::failure
+#include <stdexcept> //std::invalid_argument
+
+extern "C" {
```

Comment:
I just took a look at my copy of `bzlib.h`. On line 25, it does its own `extern "C"` work; so ostensibly there's no cause to do our own.